### PR TITLE
AYS-142 | Forbid FirstName & LastName To Start With Special Characters

### PR DESCRIPTION
**Pull request by MenekseYuncu**
We have revised Regex so that the firstname and the lastname cannot start with special characters.

### Checklist

Before submitting your pull request, ensure the following:

- [x] **Title and Branch Naming Conventions:**
    - The pull request title follows the
      standard: [Pull Request Naming Conventions](CONTRIBUTING.md#pull-request-naming-conventions).
    - The branch name follows one of the conventions outlined in
      the [Branch Naming Conventions](CONTRIBUTING.md#branch-naming-conventions).

- [x] **Local Testing:**
    - I have tested my changes locally on Postman, and they are working as expected.

- [x] **Code Quality:**
    - The code is formatted according to the project's coding guidelines and style.
    - The code has been reviewed to ensure its quality.
    - The code does not contain any issues flagged by SonarLint.

- [x] **Documentation:**
    - Necessary documentation has been added or existing documentation has been updated, specifically detailing changes
      made in Postman.

- [x] **Testing:**
    - Relevant unit tests have been written and included.
    - Relevant integration tests have been written and included.

- [x] **Reviewers and Assignees:**
    - Default reviewers have been assigned to this pull request.
    - Assignees have been added if necessary.

- [x] **Labels and Associations:**
    - No specific actions are required in the Labels and Associations section for this pull request.

- [x] **Jira Issue Transition:**
    - If this pull request is related to a Jira issue, ensure that the issue is moved to the "In Review" status on the
      Jira board.

### CODE REVIEW

**Review by agitrubard — CHANGES_REQUESTED** [review]
Buradaki validasyonu sağlayan testler ekleyebilir miyiz?

**Comment by AhmetAksunger** [comment]
> Buradaki validasyonu sağlayan testler ekleyebilir miyiz?

Ekledik, inceler misin?

File: src/main/java/com/ays/common/util/validation/NameValidator.java [hunk 1]
```diff
@@ -20,10 +20,11 @@ class NameValidator implements ConstraintValidator<Name, String> {
...
      *   <li>Common punctuation: space( ), comma(,), period(.), apostrophe('), hyphen(-)</li>
      * </ul>
+     *  It also avoids strings that start with special characters
      * </p>
      * <p>
      */
-    static final String NAME_REGEX = "^[a-zA-ZÇçĞğİıÖöŞşÜü ,.'-]+$";
+    static final String NAME_REGEX = "^(?![^a-zA-ZÇçĞğİıÖöŞşÜü]).*[a-zA-ZÇçĞğİıÖöŞşÜü ,.'-]+$";
     static final Integer NAME_MIN_LENGTH = 2;
     static final Integer NAME_MAX_LENGTH = 255;
```
Comment (agitrubard) [inline src/main/java/com/ays/common/util/validation/NameValidator.java:29]: Eğer buradaki değişkenleri başka yerlerde kullanmıyorsak `private` access modifier kullanabilir miyiz

**Review by agitrubard — APPROVED** [review]
@MenekseYuncu @AhmetAksunger Elinize sağlık, parameterized testler bayağı iyi olmuş 🚀

**Review by anilhazar — DISMISSED** [review]

**Pull request merged**